instrumentation/github.com/gin-gonic/gin/otelgin/gintrace.go

```diff
@@ -83,6 +84,15 @@ func Middleware(service string, opts ...Option) gin.HandlerFunc {
 		ctx, span := tracer.Start(ctx, spanName, opts...)
 		defer span.End()
 
+		// support recovery
```

Comment:
None of our instrumentation libraries handles panics. _Exception: `otelgin.HTML`, which does not look right to me._
I am against adding such functionality by default as users may have their own panic recovery. 
This code would change the callstack of the panic received by the next panic recovery making it hard to find the main source of the error in their existing panic recovery solution.

Reply:
I was also thinking today whether to deprecate: `otelgin.HTML` in future versions.

Reply:
I am wondering if it will cause a `panic`, resulting in the span not end yet.

Reply:
> I am wondering if it will cause a `panic`, resulting in the span not end yet.

It should end as there is a `defer span.End()`.

Reply:
> I was also thinking today whether to deprecate: `otelgin.HTML` in future versions.

Works for me

Reply:
We also recover in `span.End()`.
https://github.com/open-telemetry/opentelemetry-go/blob/8b510c36ed26fafe151df35a251f4589f5959802/sdk/trace/span.go#L420

So any panic happening should already be caught by the `defer span.End()`.

Reply:
OK, I understand. Then this PR should be closed. Thank you all very much.